src/main/scala/analysis/InterLiveVarsAnalysis.scala

```diff
@@ -0,0 +1,115 @@
+package analysis
+
+import analysis.solvers.BackwardIDESolver
+import ir.{Assert, Assume, GoTo, CFGPosition, Command, DirectCall, IndirectCall, LocalAssign, MemoryAssign, Procedure, Program, Variable, toShortString}
+
+/**
+ * Micro-transfer-functions for LiveVar analysis
+ * this analysis works by inlining function calls (instead of just mapping parameters and returns all
+ * live variables (registers) are propagated to and from callee functions)
+ * The result of what variables are alive at each point in the program should still be correct
+ * However, the functions that are callees of other functions will have an over approximation of their parameters
+ * alive at the top of the function
+ * Tip SPA IDE Slides include a short and clear explanation of microfunctions
+ * https://cs.au.dk/~amoeller/spa/8-distributive.pdf
+ */
+trait LiveVarsAnalysisFunctions extends BackwardIDEAnalysis[Variable, FlatElement[Nothing] ,TwoElementLattice] {
+
+  val valuelattice: TwoElementLattice = TwoElementLattice()
+  val edgelattice: EdgeFunctionLattice[FlatElement[Nothing], valuelattice.type] = new EdgeFunctionLattice[FlatElement[Nothing], valuelattice.type](valuelattice)
```

Comment:
There are a lot of these `FlatElement[Nothing]` and `TwoElementLattice` (which extends `FlatLattice[Nothing] floating around - are they to represent a lattice that only has two values, Top and Bottom? There should be new types created for that instead of awkwardly trying to repurpose `FlatElement` and `FlatLattice`